File: examples/async/okx_ws_public.py
"""Read OKX public WebSocket trade events."""

import asyncio

from dcex.ws import okx


async def main() -> None:
    """Subscribe to OKX trades and print two messages."""
    async with okx.public() as ws:
        await ws.subscribe_trades("BTC-USDT-SPOT")
        print(await ws.recv())
        print(await ws.recv())


if __name__ == "__main__":
    asyncio.run(main())

File: crates/dcex/examples/okx_ws_public.rs
use std::time::Duration;

use dcex::ws::okx::OkxPublicWebSocket;

#[tokio::main]
async fn main() -> dcex::Result<()> {
    let mut ws = OkxPublicWebSocket::new(Duration::from_secs(10))?;
    ws.connect().await?;
    ws.subscribe_trades("BTC-USDT-SPOT").await?;
    println!("{}", ws.recv().await?);
    println!("{}", ws.recv().await?);
    ws.close().await?;
    Ok(())
}
